src/inc/msquic.h

```diff
@@ -1334,31 +1338,31 @@ MsQuicOpenVersion(
 // Version specific helpers that wrap MsQuicOpenVersion.
 //
 
-#ifndef QUIC_CORE_INTERNAL
-
-/*#if defined(__cplusplus) || defined(WIN32)
+#if defined(__cplusplus)
 
 _IRQL_requires_max_(PASSIVE_LEVEL)
+_Check_return_
+#if (__cplusplus >= 201703L || _MSVC_LANG >= 201703L)
+[[nodiscard]]
+#endif
 #ifdef WIN32
 __forceinline
 #else
 __attribute__((always_inline)) inline
 #endif
 QUIC_STATUS
-MsQuicOpen(
+MsQuicOpen2(
     _Out_ _Pre_defensive_ const QUIC_API_TABLE** QuicApi
     )
 {
-    return MsQuicOpenVersion(1, (const void**)QuicApi);
+    return MsQuicOpenVersion(2, (const void**)QuicApi);
 }
 
-#else*/
-
-#define MsQuicOpen(QuicApi) MsQuicOpenVersion(1, (const void**)QuicApi)
+#else
 
-/*#endif // defined(__cplusplus) || defined(WIN32)*/
+#define MsQuicOpen2(QuicApi) MsQuicOpenVersion(2, (const void**)QuicApi)
```

Comment:
Why is this alias needed?

Reply:
Preference. Not strictly required.

Reply:
Also, assuming youre calling from C++ you'll get the correct output type, instead of just being void**